Implement JetSetllly and scorpheus improvements

neclepsio opened the pull request:
- GetScroll* and SetScroll*
- drag and drop
- color styles
- ContentRegionMax
- CollapsingHeader
- BeginPopup
- ItemSpacing
- two custom widgets, Spinner and Splitter, in Utils* files.

neclepsio commented:
Sorry, it was premature. For example, CollapsingHeader signature is obsolete. I will resubmit it afet some testing.

dertseha commented on imgui.go:
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
+// type is a user defined string of maximum 32 characters.
+// Strings starting with '_' are reserved for dear imgui internal types.
+// Data is copied and held by imgui.
+func SetDragDropPayload(type_ string, payload string, cond Condition) bool {
```
In the library, the payload (called `data` in the API by the way) is handled rather as an arbitrary byte array. It should be exposed as such here as well.

dertseha commented on imgui.go:
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
+func SetDragDropPayload(type_ string, payload string, cond Condition) bool {
+	typeArg, typeFin := wrapString(type_)
+	defer typeFin()
+	out := payload + "\000" + strings.Repeat(" ", 512-len(payload))
```
This is extra magic that misses explanation and I suspect this is type-specific. If the `data` is handled as arbitrary byte array, this extra magic goes to the caller.
This also includes the constant of 512 as the internal argument. Please rework.

dertseha commented on imgui.go:
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
+
+// AcceptDragDropPayload accepts contents of a given type.
+// If ImGuiDragDropFlags_AcceptBeforeDelivery is set you can peek into the payload before the mouse button is released.
+func AcceptDragDropPayload(type_ string, flags int) string {
```
Rework this function as well, please. Return type []byte and the internal functions need reworking for this.
(This then also removes the hardcoded 512).

neclepsio commented:
Ok, as I wrote in the description I simply merged scorpheus improvements. I just started using ImGui and your wrapper. so please excuse if something I make big errors. As for comments as proper sentences, I read the contributing guidelines, and rewrote many of the comments, but missed some. 

P.S.: would you accept a refactoring that, for example, moved the `Column*` functions to new .go/.cpp/.h files? I noticed that the majority of functions are in `imgui.go` but, for example, you separated some in `Style.go`.

dertseha commented:
All good - I appreciate your effort to collect some extensions that others made and didn't propose.
When adding features upstream, then additions need more consideration than were perhaps necessary for a specific fork.
You can still add some of the extensions if you don't want to spend extra effort on all the fixes.

It may be useful to extract some feature-blocks into separate files, yes. `imgui.go` started as the general bin for all the "global" functions, yet this does not scale well.
Unless it's a minor change, I recommend doing such refactorings as dedicated pull requests.
